tests/snapshot/snapshot.go

```diff
@@ -0,0 +1,401 @@
+package snapshot
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	extdefault "github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/etcdsnapshot"
+	"github.com/rancher/shepherd/extensions/ingresses"
+	nodestat "github.com/rancher/shepherd/extensions/nodes"
+	"github.com/rancher/shepherd/extensions/provisioning"
+	"github.com/rancher/shepherd/extensions/workloads"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/rancher/tfp-automation/config"
+	set "github.com/rancher/tfp-automation/framework/set"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networking "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	all                          = "all"
+	containerImage               = "nginx"
+	containerName                = "nginx"
+	defaultNamespace             = "default"
+	DeploymentSteveType          = "apps.deployment"
+	isCattleLabeled              = true
+	IngressSteveType             = "networking.k8s.io.ingress"
+	ingressPath                  = "/index.html"
+	initialIngressName           = "ingress-before-restore"
+	initialWorkloadName          = "wload-before-restore"
+	localClusterName             = "local"
+	K3S                          = "k3s"
+	kubernetesVersion            = "kubernetesVersion"
+	namespace                    = "fleet-default"
+	port                         = "port"
+	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
+	RKE1                         = "rke1"
+	RKE2                         = "rke2"
+	serviceAppendName            = "service-"
+	ServiceType                  = "service"
+	WorkloadNamePostBackup       = "wload-after-backup"
+)
+
+func snapshotRestore(t *testing.T, client *rancher.Client, clusterName string, clusterConfig *config.TerratestConfig, terraformOptions *terraform.Options) {
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(t, err)
+
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(t, err)
+
+	localClusterID, err := clusters.GetClusterIDByName(client, localClusterName)
+	require.NoError(t, err)
+
+	var isRKE1 bool
```

Comment:
super nit; easier to read if you set a default value for this i.e. `false` and only have an if statement instead of if/else.

Reply:
I see what you're saying here. Would it be too much trouble if we make the change in rancher/shepherd first and then here? Just to have them all on parity.